Parsers/ASimWebSession/Parsers/ASimWebSessionBarracudaWAF.yaml

```diff
@@ -0,0 +1,305 @@
+                                CacheHit_d == 0,
+                                "Response from the server",
+                                "Response from the cache"
+                            ),
+          AdditionalFields = bag_pack(
+                        "ProxyIP",
+                        ProxyIP_s,
+                        "ProxyPort",
+                        ProxyPort_d
+                    ),
+          DvcHostname = host_s,
+          DvcIpAddr = HostIP_s,
+          EventResultDetails = HTTPStatus_s
+      | extend 
+          SrcUsernameType = iff(isnotempty(SrcUsername), "Simple", ""),
+          DstUsernameType = iff(isnotempty(DstUsername), "Simple", "")
+      | extend
+          Duration = NetworkDuration,
+          SessionId = NetworkSessionId,
+          EventEndTime = EventStartTime,
+          UserAgent = HttpUserAgent,
+          User = SrcUsername,
+          IpAddr = SrcIpAddr,
+          Src = SrcIpAddr,
+          HttpStatusCode = EventResultDetails
+      | project-away
+          *_d,
+          *_s,
+          _ResourceId,
+          severity,
+          status_code,
+          RawData,
+          EventResult_*,
+          SourceIP,
+          Message,
+          EventType_lookup,
+          Computer,
+          MG,
+          ManagementGroupName,
+          TenantId,
+          SourceSystem
+      )
+  };
+  let BarracudaCEF = (disabled: bool=false) {
+      CommonSecurityLog
+      | where not(disabled) and DeviceVendor == "Barracuda" and DeviceProduct == "WAF"
+      | where DeviceEventCategory in ("WF", "TR")
+      | lookup EventResultWFLookup on $left.DeviceAction == $right.Action_s
+      | lookup EventTypeLookup on $left.DeviceEventCategory == $right.LogType_s
+      | extend
+          EventType = EventType_lookup,
+          severity = toint(LogSeverity)
+      | lookup SeverityLookup on severity
+      | extend
+          Dst = DestinationIP,
+          EventCount = toint(1),
+          EventProduct = "WAF",
+          EventSchema = "WebSession",
+          EventSchemaVersion = "0.2.6",
+          EventVendor = "Barracuda",
+          status_code = toint(EventOutcome)
+      | extend
+          EventResult_TR = case(
+                      status_code between (200 .. 299),
+                      "Success", 
+                      status_code between (400 .. 599),
+                      "Failure",
+                      status_code between (300 .. 399),
+                      "Partial",
+                      "NA"
+                  ),
+          RuleName = iff(DeviceEventCategory == "WF", DeviceCustomString3, "")
+      | extend
+          Dvc = DeviceName,
+          EventResult = iff(DeviceEventCategory == "TR", EventResult_TR, EventResult_WF),
+          EventStartTime = unixtime_milliseconds_todatetime(tolong(ReceiptTime)),
+          DstIpAddr = DestinationIP,
+          SrcIpAddr = SourceIP,
+          DstBytes = tolong(ReceivedBytes),
+          DstPortNumber = toint(DestinationPort),
```

Comment:
This value is present in DeviceCustomNumber1 as well. Please populate from there as well.

Reply:
Done